# Data exploration

We'll begin by reading in the data

In [33]:
import pandas as pd
df = pd.read_csv("data/credit-data.csv")

Checking the column names

In [44]:
df.columns

Index(['PersonID', 'SeriousDlqin2yrs', 'RevolvingUtilizationOfUnsecuredLines',
       'age', 'zipcode', 'NumberOfTime30-59DaysPastDueNotWorse', 'DebtRatio',
       'MonthlyIncome', 'NumberOfOpenCreditLinesAndLoans',
       'NumberOfTimes90DaysLate', 'NumberRealEstateLoansOrLines',
       'NumberOfTime60-89DaysPastDueNotWorse', 'NumberOfDependents'],
      dtype='object')

Now we'll check the type of each column

In [31]:
for col in df.columns:
    print(col, 'is type --->', df[col].dtype)

PersonID is type ---> int64
SeriousDlqin2yrs is type ---> int64
RevolvingUtilizationOfUnsecuredLines is type ---> float64
age is type ---> int64
zipcode is type ---> int64
NumberOfTime30-59DaysPastDueNotWorse is type ---> int64
DebtRatio is type ---> float64
MonthlyIncome is type ---> float64
NumberOfOpenCreditLinesAndLoans is type ---> int64
NumberOfTimes90DaysLate is type ---> int64
NumberRealEstateLoansOrLines is type ---> int64
NumberOfTime60-89DaysPastDueNotWorse is type ---> int64
NumberOfDependents is type ---> float64


Some columns have float types when they should have int, but for our exploration purposes, all variables have an appropriate numeric type.

Checking the proportion of missing values:

In [46]:
total = len(df)
for col in df.columns:
    print(col, 'has', df[col].isna().sum() / total, 'as missings proportion')

PersonID has 0.0 as missings proportion
SeriousDlqin2yrs has 0.0 as missings proportion
RevolvingUtilizationOfUnsecuredLines has 0.0 as missings proportion
age has 0.0 as missings proportion
zipcode has 0.0 as missings proportion
NumberOfTime30-59DaysPastDueNotWorse has 0.0 as missings proportion
DebtRatio has 0.0 as missings proportion
MonthlyIncome has 0.1944119368051492 as missings proportion
NumberOfOpenCreditLinesAndLoans has 0.0 as missings proportion
NumberOfTimes90DaysLate has 0.0 as missings proportion
NumberRealEstateLoansOrLines has 0.0 as missings proportion
NumberOfTime60-89DaysPastDueNotWorse has 0.0 as missings proportion
NumberOfDependents has 0.02528281646186854 as missings proportion


`MonthlyIncome` has 19.4% of observations with missing values and `NumberOfDependents` has 2.5%. This will cause us to lose a non-trivial number of observations when conducting our analysis. To avoid this, we'll have to conduct some kind of imputation rule in these cases later on.

Tabulating the target attribute:

In [15]:
df.groupby('SeriousDlqin2yrs').size()

SeriousDlqin2yrs
0    34396
1     6620
dtype: int64

We see that our dataframe has 34396 + 6620 = 41016 observations. 16% of observations have a value of one for the variable `SeriousDlqin2yrs`.

Now we'll graph a correlation matrix color map to know which of the variables we have are correlated with the target attribute.

In [36]:
df.corr().style.background_gradient(cmap='coolwarm')

Surprisingly, the variable with the highest correlation with `SeriousDlqin2yrs` is `PersonID`. As the data dictionary provided doesn't include any explanation of how the ID was assigned, we'll assume this correlation is spurious. Appart from that, the variable with the highest correlation with `SeriousDlqin2yrs` is `age`. In general, none variable has a remarkably high correlation with our target attribute, and all of them seem to have similarly low correlation levels.